### PR TITLE
feat: Add push-artifacthub action, push artifacthub files to artifacthub branch 

### PR DESCRIPTION
**Pull request by viccuad**
## Description

Relates to https://togithub.com/artifacthub/hub/issues/2922

## Test

Tested from viccuad@github-actions
and consumed by https://github.com/viccuad/allowed-fsgroups-psp-policy

See these 4 runs for main and 0.1.9:
- triggered by main, test.yml workflow: https://github.com/viccuad/allowed-fsgroups-psp-policy/actions/runs/4502885701
  no changes
- triggered by main, relese.yml workflow: https://github.com/viccuad/allowed-fsgroups-psp-policy/actions/runs/4502885737
  Creates the `:latest` image. Skips check-artifachub step, as versions may be different, skips push-artifacthub job, as we don't want to commit the artifacthub-pkg.yml changes.
- triggered by 0.1.9 tag, test.yml worfklow: https://github.com/viccuad/allowed-fsgroups-psp-policy/actions/runs/4502950922
  no changes
- triggered by 0.1.9 tag, release.yml workflow: https://github.com/viccuad/allowed-fsgroups-psp-policy/actions/runs/4502950913
  Creates the `:v0.1.9` image, does check-artifacthub, does push-artifacthub

The same for runs for main and 0.1.10.

The repo has an `artifacthub` branch:
https://github.com/viccuad/allowed-fsgroups-psp-policy/tree/artifacthub

The policy appears signed in artifacthub:
https://artifacthub.io/packages/viccuad/test-policy/allowed-fsgroups-psp
there, version 0.1.9 is missing metadata & readme because I forgot to update artifacthub-repo.yml with ID.
Version 0.1.10 should all be fine.
Next check for 0.1.10 will happen at 18:34 UTC approx. In 20 mins after this PR was open.

### CODE REVIEW

File: push-artifacthub/action.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      run: |
+        git config user.name "Update artifacthub branch"
+        git config user.email github-actions@github.com
+
+        git checkout artifacthub || git checkout --orphan artifacthub
+        git reset HEAD -- .
+        git add artifacthub-pkg.yml artifacthub-repo.yml
+        VERSION=$(sed --posix -n 's,^version: \(.*\),\1,p' artifacthub-pkg.yml)
+        git commit -m "Update Artifact Hub files, version $VERSION"
+        git push --force origin artifacthub
```
Comment (flavio) [inline push-artifacthub/action.yml:24]: Using an orphan branch simplifies everything, but is artifacthub going to be fine with it?

**Review by flavio — APPROVED** [review]
overall ok, I just have one doubt. See comment

File: push-artifacthub/action.yml (hunk 1 above)
Comment (viccuad) [reply push-artifacthub/action.yml:24]: it seems to be ok with an orphan branch, it finds the `artifacthub-pkg.yml` and the `artifacthub-repo.yml`.

The only problem I see, is missing the README.md. [artifacthub-pkg.yml `readme` field, if missing, gets read from the README.md file](https://github.com/artifacthub/hub/blob/88aae95607ca1905ee93059a55601e4635732678/docs/metadata/artifacthub-pkg.yml#L31).

I will `git add README.md` too.

Comment (viccuad) [reply push-artifacthub/action.yml]: Done in https://github.com/kubewarden/github-actions/pull/69/commits/a53058f1efc5f70dddb67b60d3fb4b7a824ffd9c.

**Comment by viccuad** [comment]
tagged as v3.0.5

**Comment by viccuad** [comment]
deleted tag v3.0.5, tagged as v3.1.0, with a commit that consumes v3.1.0 too.

**Pull request merged**